### PR TITLE
Add kubernetes logging blog post

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+It is crucial in this scenario to configure any logging agent's state to be written to disk on the host, so that if the logging agent pod restarts, it will resume from its last recorded position avoiding lost logs and duplicated logs.
+
+Link to highlighted lines?
+
+The Not So Happy Path
+
+So you still have a Kubernetes cluster up and running with containerized applications, but they are logging disk. This situation is not uncommon, and fortunately, we can handle it.
+
+While one approach is forwarding the logs to stdout and error (https://serverfault.com/questions/599103/make-a-docker-application-write-to-stdout/634296#634296), we would then be on the Happy Path. Instead, we will assume the logs will always be written to disk and work around it.
```
Comment (zsherman) [inline posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md]: Don't need "While"

File: posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+To demonstrate our solution, we will provide examples for getting the logs out of an nginx container.
+
+_NOTE: The following examples should work on any kube cluster, including minikube_
+
+Link to nginx pod
```
Comment (zsherman) [inline posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md]: These all need to be filled out, right?

File: posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+kubectl command to create nginx pod
+
+Our first strategies here involve minimal changes, leveraging the DaemonSet based solution from The Happy Path.
```
Comment (zsherman) [inline posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md]: "strategy"

File: posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+The one downside to this approach is that the nginx log files now written to the host file system will have to be rotated, or else we risk the chance of filling up the host disk space potentially making it unhealthy.
+
+2. Streaming files to stdout and stderr of a sidecar container
+
+As highlighted by the Kubernetes documentation, another way to retrieve the logs of a file in a container is to write to a shared volume, and stream the logs to stdout and stderr of a sidecar container. This strategy also leverages our existing logging agent, as the logs from the sidecar container will be written under `/var/log/containers/`.
+
+link to nginx with sidecar streamer
+kubectl command
+
+We can see our side car working via the following:
```
Comment (zsherman) [inline posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md]: "side car" or "sidecar"? Should be consistent.

File: posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
+Moving Application Logs out of Kubernetes (and into your Workflow)
```
Comment (zsherman) [inline posts/moving-application-logs-out-of-kubernetes-and-into-your-workflow.md]: Seems like you're missing the markdown headers/styling in this posts.

**Comment by zsherman** [comment]
Overall I still don't quite walk away from this feeling productive or having a great understanding of how to pull this off. Seems like we're lacking in code samples (which you probably can pull from the example repo) and a diagram wouldn't be terrible either, the [Kub docs](https://kubernetes.io/docs/concepts/cluster-administration/logging/) have some good ones so you can either embed those or draw some simple ones.

**Comment by zsherman** [comment]
Also I don't think it would be terrible to have a section on what specifically you want to be capturing in Kub, what the default logs look like, etc.

**Comment by ghost** [comment]
Agree 100% on this. I wrapped up some examples yesterday and that will be driving the discussion/content of the post. More of a run these commands, you should see this, and let's talk about what it means.

**Comment by ghost** [comment]
@zsherman I think we scrape the making logs better for another post. Just feel like I'm introducing a new topic, and I think it deserves a good amount of explanation. We could talk about structured logging, maximizing contextual data, and what that looks like in a Kube setting. Also talk about Kube controllers and other resources. Can easily get 1k lines out of that.

**Comment by zsherman** [comment]
Sure, we can save it for a part II. I think this most recent update is a significant improvement.

**Pull request merged**